MATLAB (#76)
ubccr/ccr-examples: containers/2_ApplicationSpecific/MATLAB/README.md

# Example MATLAB containers

MATLAB is a suite of commercial prorgams from [MathWorks](https://www.mathworks.com/products/matlab.html)  
CCR has a license for academic use of serveral MATLAB programs, but you MUST meet MathWorks eligability contraints to use the CCR license.  

## Building the container

A brief guide to building the MATLAB container follows:  
Please refer to CCR's [container documentation](https://docs.ccr.buffalo.edu/en/latest/howto/containerization/) for more detailed information on building and using Apptainer.

1. Start an interactive job

Apptainer is not available on the CCR login nodes and the compile nodes may not provide enough resources for you to build a container.  We recommend requesting an interactive job on a compute node to conduct this build process.  
Note: a GPU is NOT needed to build the MATLAB container  
See CCR docs for more info on [running jobs](https://docs.ccr.buffalo.edu/en/latest/hpc/jobs/#interactive-job-submission)

```
export SBATCH_ACCOUNT="[SlurmAccountName]"
```

```
salloc --cluster=ub-hpc --partition=debug --qos=debug --mem=0 --exclusive \
 --time=01:00:00
```

> sample outout:

Reviewer: ```suggestion
Sample outout:
```

> ```
> salloc: Granted job allocation 25260020
> salloc: Nodes cpn-b04-32-01 are ready for job
> CCRusername@cpn-b04-32-01$ 
> ```

2. Navigate to your build directory and use the Slurm job local temporary directory for cache

You should now be on the compute node allocated to you.  
In this example we're using our project directory for our build directory.  

Change to your MATLAB directory

```
cd /projects/academic/[YourGroupName]/MATLAB
```

We have provided two different build examples:  
A parallel minimal install with only the "MATLAB" and "Parallel Computing Toolbox" products  
A maximal install, with all the products for which CCR currenlty has a license

If you wish to install a specific subset of products, download the maximal

> install example and remove the products you do not need from the "

Reviewer: There is no information after the quotation marks.

> install example and remove the products you do not need from the "

Author: ```suggestion
install example and remove the products you do not need from the "MATLAB-R2025b.def" file
```

# Example MATLAB containers

MATLAB is a suite of commercial prorgams from [MathWorks](https://www.mathworks.com/products/matlab.html)  

> CCR has a license for academic use of serveral MATLAB programs, but you MUST meet MathWorks eligability contraints to use the CCR license.  

Reviewer 2: ```suggestion
> [!WARNING]
> UB has a license for academic use of several MATLAB programs, but you MUST meet MathWorks eligibility constraints to use the UB license.  Academic use is restricted to non-commercial teaching, learning, and academic research and cannot be used for any commercial or revenue-generating work.  This license is NOT for use by Roswell Park researchers or any commercial business or start-up utilizing CCR's industry cluster.
```

Apptainer is not available on the CCR login nodes and the compile nodes may not provide enough resources for you to build a container.  We recommend requesting an interactive job on a compute node to conduct this build process.  
Note: a GPU is NOT needed to build the MATLAB container  
See CCR docs for more info on [running jobs](https://docs.ccr.buffalo.edu/en/latest/hpc/jobs/#interactive-job-submission)

```bash
export SBATCH_ACCOUNT="[SlurmAccountName]"
```

```bash
salloc --cluster=ub-hpc --partition=debug --qos=debug --mem=0 --exclusive \
 --time=01:00:00
```

Sample outout:

> ```
> salloc: Granted job allocation 25260020
> salloc: Nodes cpn-b04-32-01 are ready for job
> CCRusername@cpn-b04-32-01$ 
> ```

2. Navigate to your build directory and use the Slurm job local temporary directory for cache

You should now be on the compute node allocated to you.  
In this example we're using our project directory for our build directory.  

Change to your MATLAB directory

```bash
cd /projects/academic/[YourGroupName]/MATLAB
```

We have provided two different build examples:  
A parallel minimal install with only the "MATLAB" and "Parallel Computing Toolbox" products  

> A maximal install, with all the products for which CCR currenlty has a license

Reviewer 2: ```suggestion
A maximal install, with all the products for which UB currently has a license
```

> Feature usage info:
> 
> Users of MATLAB_Distrib_Comp_Engine:  (Total of 100000 licenses issued;  Total of 0 licenses in use)
> 
> Users of MATLAB:  (Total of 10000 licenses issued;  Total of 3 licenses in use)
> 
>   
> "MATLAB" v54, vendor: MLM, expiry: 30-oct-2026
>   vendor_string: vi=0:at=200:ae=1:lu=200:lo=TH:ei=1233456:lr=1:ep=7:
>   floating license
> 
>     user1 cpn-b01-08-02.core.ccr.buffalo.edu /dev/pts/0 (v50) (license1.ccr.buffalo.edu/27000 16636), start Tue 7/14 23:35, PID: 3278807 
>     user2 cpn-d01-32.core.ccr.buffalo.edu /dev/pts/0 (v50) (license1.ccr.buffalo.edu/27000 40623), start Fri 7/17 10:20, PID: 2636405 
>     user2 cpn-d02-13.core.ccr.buffalo.edu /dev/pts/0 (v50) (license1.ccr.buffalo.edu/27000 17527), start Fri 7/17 10:58, PID: 56371 
> 
> Users of SIMULINK:  (Total of 10000 licenses issued;  Total of 0 licenses in use)
> 
> Users of MATLAB_5G_Toolbox:  (Total of 10000 licenses issued;  Total of 0 licenses in use)
> [...]
> Users of Wireless_Testbench:  (Total of 10000 licenses issued;  Total of 0 licenses in use)
> 
> ```

See the [EXAMPLES file](./EXAMPLES.md) for more info.  

## Sample Slurm scripts

### Parallel example:

> [MATLAB parallel Slurm example script](https://raw.githubusercontent.com/ubccr/ccr-examples/refs/heads/main/containers/2_ApplicationSpecific/MATLAB/matlab-mp.bash)  

Reviewer 2: ```suggestion
[MATLAB parallel Slurm example script](matlab-mp.bash)  
```
I think removing the full path will allow these links to work since they're all in the same directory

### GPU example:

> [MATLAB GPU Slurm example script](https://raw.githubusercontent.com/ubccr/ccr-examples/refs/heads/main/containers/2_ApplicationSpecific/MATLAB/matlab-gpu.bash)  

Reviewer 2: ```suggestion
[MATLAB GPU Slurm example script](matlab-gpu.bash)  
```
